clippy_lints/src/floating_point_arithmetic.rs

```diff
@@ -324,7 +324,11 @@ fn check_powi(cx: &LateContext<'_>, expr: &Expr<'_>, receiver: &Expr<'_>, args:
                     let maybe_neg_sugg = |expr, hir_id| {
                         let sugg = Sugg::hir(cx, expr, "..");
                         if matches!(op, BinOpKind::Sub) && hir_id == rhs.hir_id {
-                            format!("-{sugg}")
+                            if let ExprKind::Binary(..) = expr.kind {
+                                format!("-({sugg})")
+                            } else {
+                                format!("-{sugg}")
+                            }
```

Comment:
Using `sugg.maybe_paren()` would handle more cases.

Reply:
Thanks for the advice! I should have guessed that there would be something in the `Sugg` struct that would take care of cases like this.